kcl/kcl.go

```diff
@@ -224,15 +224,15 @@ func (kclp *KCLProcess) reportDone(responseFor string) error {
 	})
 }
 
-func (kclp *KCLProcess) handleLine(line string) error {
+func (kclp *KCLProcess) handleLine(line string) (string, error) {
 	action, err := kclp.ioHandler.loadAction(line)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	switch action := action.(type) {
 	case ActionCheckpoint:
-		err = kclp.handleCheckpointAction(action)
+		return "checkpoint", kclp.handleCheckpointAction(action)
```

Comment:
can we use constants here rather than strings directly.

Reply:
I'm not sure if it's worth the overhead/code cruft to make these strings constants.  Currently, `handleLine` is a private function that's called in one place (in `Run`), and the only relevant return value is "shutdown".  I thought about making `handleLine` return a `isShutdown` flag or the `action` variable, but I felt both those options were less readable.  A naked boolean always feels ambiguous and returning `action` would require a `switch action.(type)` in `Run` which felt overly complicated.  Another possibility was to create a shutdown-specific error, but I felt that would be confusing as well.

Reply:
I appreciate the explanation but don't really agree or understand with your unilateral assessment that adding strings to be constants is code cruft.

Reply:
agree with @mohit on returning an enum type instead of a string